Signal/src/ViewControllers/ConversationView/ConversationViewLayout.m

```diff
@@ -94,60 +94,42 @@ - (void)prepareLayout
     // TODO: Remove this log statement after we've reduced the invalidation churn.
     DDLogVerbose(@"%@ prepareLayout", self.logTag);
 
-    const int vInset = 8;
-    const int hInset = 10;
-    const int vSpacing = 5;
-    const int viewWidth = (int)floor(self.collectionView.bounds.size.width);
-    const int contentWidth = (int)floor(viewWidth - 2 * hInset);
-    self.contentWidth = contentWidth;
+    const CGFloat viewWidth = self.layoutInfo.viewWidth;
 
     NSArray<id<ConversationViewLayoutItem>> *layoutItems = self.delegate.layoutItems;
 
-    CGFloat y = vInset + self.delegate.layoutHeaderHeight;
+    CGFloat y = self.layoutInfo.contentMarginTop + self.delegate.layoutHeaderHeight;
     CGFloat contentBottom = y;
-    BOOL isRTL = self.collectionView.isRTL;
 
     NSInteger row = 0;
+    id<ConversationViewLayoutItem> _Nullable lastLayoutItem = nil;
     for (id<ConversationViewLayoutItem> layoutItem in layoutItems) {
-        CGSize layoutSize = [layoutItem cellSizeForViewWidth:viewWidth contentWidth:contentWidth];
-
-        layoutSize.width = MIN(viewWidth, floor(layoutSize.width));
-        layoutSize.height = floor(layoutSize.height);
-        CGRect itemFrame;
-        switch (layoutItem.layoutAlignment) {
-            case ConversationViewLayoutAlignment_Incoming:
-            case ConversationViewLayoutAlignment_Outgoing: {
-                BOOL isIncoming = layoutItem.layoutAlignment == ConversationViewLayoutAlignment_Incoming;
-                BOOL isLeft = isIncoming ^ isRTL;
-                if (isLeft) {
-                    itemFrame = CGRectMake(hInset, y, layoutSize.width, layoutSize.height);
-                } else {
-                    itemFrame
-                        = CGRectMake(viewWidth - (hInset + layoutSize.width), y, layoutSize.width, layoutSize.height);
-                }
-                break;
-            }
-            case ConversationViewLayoutAlignment_FullWidth:
-                itemFrame = CGRectMake(0, y, viewWidth, layoutSize.height);
-                break;
-            case ConversationViewLayoutAlignment_Center:
-                itemFrame = CGRectMake(
-                    hInset + round((viewWidth - layoutSize.width) * 0.5f), y, layoutSize.width, layoutSize.height);
-                break;
+        if (lastLayoutItem) {
```

Comment:
nit: `last` is ambiguous. I think you mean `previous`?

Reply:
That's better, thanks.